torch/csrc/jit/codegen/cuda/expr_evaluator.cpp

```diff
@@ -26,15 +27,20 @@ void ExpressionEvaluator::bind(Val* value, Int::ScalarType concrete_value) {
 }
 
 c10::optional<Int::ScalarType> ExpressionEvaluator::evaluate(Val* value) {
-  FUSER_PERF_SCOPE("ExpressionEvaluator::evaluate");
-  auto maybe_concrete_value = getValue(value);
-  if (!maybe_concrete_value.has_value()) {
-    if (value->definition() != nullptr) {
-      OptOutDispatch::handle(value->definition());
-      maybe_concrete_value = getValue(value);
+  if (evaluator_precomputed_integers_ != nullptr) {
```

Comment:
Is this the difference between using evaluator for actual runtime versus for the schedulers? Uncertain when we wouldn't want to use the evaluator added here.

Reply:
Actually the new evaluator path (`PreComputedIntegers`) was used to speed up runtime only and works only when we know all the integers we will need ahead of time. The legacy evaluator uses lazy-evaluation which is quite flexible and probably wouldn't be a perf bottleneck for compile-time, so I thought we'd keep the legacy evaluation path in this PR. It'd probably make sense to unify the evaluation path as much as possible though, which I could do in follow ups.

Reply:
Yeah, I think it would be nice to remove as much of the legacy code as possible. If we still need something more flexible, then we should probably just try to simply them as much as possible.